src/flamegraph/flamegraph.js

```diff
@@ -192,13 +239,13 @@ function zoom(node) {
     var attr = find_child(node, "rect").attributes;
     var width = parseFloat(attr.width.value);
     var xmin = parseFloat(attr.x.value);
-    var xmax = parseFloat(xmin + width);
+    var xmax = xmin + width;
     var ymin = parseFloat(attr.y.value);
-    var ratio = (svg.width.baseVal.value - 2 * xpad) / width;
+    var ratio = 100 / width;
     // XXX: Workaround for JavaScript float issues (fix me)
-    var fudge = 0.0001;
+    var fudge = 0.001;
```

Comment:
What's the reasoning behind lowering the fudge factor?

Reply:
When switching to percentages, the original fudge factor didn't work well anymore. It was comparing numbers like `100.0` and `99.9996`, and not seeing them as the same. This meant a lot of rectangles wouldn't draw when they should, and vice versa.

Reply:
I guess I almost wonder whether it should be even larger, but if you've found this to work, that's good enough for me.

Reply:
This value has worked well in all of the testing I've done.